test/src/e2e_vm_tests/mod.rs

```diff
@@ -288,20 +279,51 @@ pub fn run(filter_regex: Option<regex::Regex>) {
         ),
     ];
 
-    let mut number_of_tests_run = positive_project_names.iter().fold(0, |acc, (name, res)| {
-        if filter(name) {
-            assert_eq!(crate::e2e_vm_tests::harness::runs_in_vm(name), *res);
-            // cannot use partial eq on type `anyhow::Error` so I've used `matches!` here instead.
-            // https://users.rust-lang.org/t/issues-in-asserting-result/61198/3 for reference.
-            assert!(matches!(
-                crate::e2e_vm_tests::harness::test_json_abi(name),
-                Ok(())
-            ));
-            acc + 1
-        } else {
-            acc
-        }
-    });
+    let mut number_of_tests_run =
+        positive_project_names_no_abi
+            .iter()
+            .fold(0, |acc, (name, res)| {
+                if filter(name) {
+                    assert_eq!(crate::e2e_vm_tests::harness::runs_in_vm(name), *res);
+                    acc + 1
+                } else {
+                    acc
+                }
+            });
+
+    // Programs that should successfully compile, include abi and terminate
+    // with some known state. Note that if a non-contract is included
+    // it will be rejected during assertion. Please move it to
+    // `positive_project_names_no_abi` above.
+    let positive_project_names_with_abi = vec![
+        // contracts revert because this test runs them against the VM
+        // and no selectors will match
+        (
+            "should_pass/test_contracts/contract_abi_impl",
+            ProgramState::Revert(0),
+        ),
+        (
+            "should_pass/language/valid_impurity",
+            ProgramState::Revert(0), // false
+        ),
```

Comment:
All test contracts should have their ABI checked -- just ran a quick grep for them.
```console
test_contracts/basic_storage/src/main.sw
1:contract;

test_contracts/test_fuel_coin_contract/src/main.sw
1:contract;

test_contracts/increment_contract/src/main.sw
1:contract;

test_contracts/auth_testing_contract/src/main.sw
1:contract;

test_contracts/balance_test_contract/src/main.sw
1:contract;

test_contracts/storage_access_contract/src/main.sw
1:contract;

test_contracts/context_testing_contract/src/main.sw
1:contract;

test_contracts/contract_abi_impl/src/main.sw
1:contract;

language/valid_impurity/src/main.sw
1:contract;

language/storage_declaration/src/main.sw
1:contract;
```

Reply:
In this PR I just took what was already there in the `positive_project_names` and separated them to make the test that was previously there, pass.

Reply:
Opened #1124 since this PR already encompasses so much

Reply:
Contracts don't actually run and return a value, so they all have a constant return state. It is invariable. 

We need to test each of their ABIs though -- before this PR, we were checking all of those ABIs via the ABI tests. This PR removed  the ABI tests for _everything_ and only re-added them for two contracts, when in reality there are quite a few more. I think we should be able to keep those ABI tests in this PR, otherwise we would be losing test coverage.

Reply:
>This PR removed the ABI tests for everything and only re-added them for two contracts, when in reality there are quite a few more. 

I only separated the ABI tests that were a part of `positive_project_names`, the rest of the contracts listed here didn't change. If they aren't being tested now, they weren't being tested before?

Reply:
Ah, I see. This is actually a significant test gap and regression in test coverage that previously went undetected. We have not been testing the ABIs of anything that actually _has_ an ABI...eh....

Ok, could you tackle #1124 after this in a fast-follow PR then? That's a pretty important test to have.